### PR TITLE
feat: detect already updated lock file updates

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Adds detection for already updated lock file updates to bundler and composer.

## Context:

Composer - #13253

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/manager/bundler/update-locked.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import { UpdateLockedConfig, UpdateLockedResult } from '../types';
+import { extractLockFileEntries } from './locked-version';
+
+export function updateLockedDependency(
+  config: UpdateLockedConfig
+): UpdateLockedResult {
+  const { depName, currentVersion, newVersion, lockFile, lockFileContent } =
+    config;
+  logger.debug(
+    `bundler.updateLockedDependency: ${depName}@${currentVersion} -> ${newVersion} [${lockFile}]`
```
Comment (viceice) [inline lib/manager/bundler/update-locked.ts:11]: should we use logger metadata here?

File: lib/manager/composer/update-locked.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { logger } from '../../logger';
+import { api as composer } from '../../versioning/composer';
+import { UpdateLockedConfig, UpdateLockedResult } from '../types';
```
Comment (viceice) [inline lib/manager/composer/update-locked.ts:3]:
```suggestion
import type { UpdateLockedConfig, UpdateLockedResult } from '../types';
```

File: lib/manager/composer/update-locked.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import { ComposerLock } from './types';
```
Comment (viceice) [inline lib/manager/composer/update-locked.ts:4]:
```suggestion
import type { ComposerLock } from './types';
```

File: lib/manager/composer/update-locked.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+export function updateLockedDependency(
+  config: UpdateLockedConfig
+): UpdateLockedResult {
+  const { depName, currentVersion, newVersion, lockFile, lockFileContent } =
+    config;
+  logger.debug(
+    `composer.updateLockedDependency: ${depName}@${currentVersion} -> ${newVersion} [${lockFile}]`
```
Comment (viceice) [inline lib/manager/composer/update-locked.ts:12]: see comment above

**Review by viceice — COMMENTED** [review]
strange test failure 🤔

File: lib/manager/bundler/update-locked.ts (hunk 1 above)
Comment (rarkins) [reply lib/manager/bundler/update-locked.ts:11]: I think this line is ok, we already have it for others

File: lib/manager/bundler/update-locked.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import { logger } from '../../logger';
+import { UpdateLockedConfig, UpdateLockedResult } from '../types';
```
Comment (viceice) [inline lib/manager/bundler/update-locked.ts:2]:
```suggestion
import type { UpdateLockedConfig, UpdateLockedResult } from '../types';
```

**Review by viceice — APPROVED** [review]

**Pull request merged**